internal/fs/node_unix.go

```diff
@@ -9,8 +9,17 @@ import (
 	"github.com/restic/restic/internal/restic"
 )
 
-func lchown(name string, uid, gid int) error {
-	return os.Lchown(name, uid, gid)
+func lchown(name string, node *restic.Node, lookupByName bool) error {
```

Comment:
This definitely needs a test.

Reply:
We definitely don't want to require running the tests as root. But a basic sanity check should be possible. A user is is allowed to chown files owned by them to their own uid. This is effectively a no-op, but should be enough to sanity check the overall username / uid translation steps.

Other than that the tests in `internal/fs/node_test.go` should also be run with `lookupByName=true` at least once. There will be quite a few corner cases which won't be covered by those tests, but a basic smoke test is better than nothing at all.